cartographer/mapping/internal/2d/overlapping_submaps_trimmer_2d.cc

```diff
@@ -160,27 +168,30 @@ std::vector<SubmapId> FindSubmapIdsToTrim(
   }
 
   DCHECK(std::is_sorted(submap_ids_to_keep.begin(), submap_ids_to_keep.end()));
+
+  std::vector<SubmapId> result;
   std::set_difference(all_submap_ids.begin(), all_submap_ids.end(),
                       submap_ids_to_keep.begin(), submap_ids_to_keep.end(),
-                      result.begin());
+                      std::inserter(result, result.begin()));
```

Comment:
I assume this does not lead to n expensive inserts to the front of a vector.

Reply:
good point, changed